src/main/java/core/basesyntax/service/FruitShopServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package core.basesyntax.service;
+
+import core.basesyntax.converter.DataConverter;
+import core.basesyntax.file.reader.FileReader;
+import core.basesyntax.file.writer.FileWriter;
+import core.basesyntax.handler.OperationHandler;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.report.ReportGenerator;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.List;
+
+public class FruitShopServiceImpl implements FruitShopService {
+    private final FileReader fileReader;
+    private final FileWriter fileWriter;
+    private final DataConverter dataConverter;
+    private final OperationStrategy operationStrategy;
+    private final ReportGenerator reportGenerator;
+
+    public FruitShopServiceImpl(FileReader fileReader,
+                                FileWriter fileWriter,
+                                DataConverter dataConverter,
+                                OperationStrategy operationStrategy,
+                                ReportGenerator reportGenerator) {
+        this.fileReader = fileReader;
+        this.fileWriter = fileWriter;
+        this.dataConverter = dataConverter;
+        this.operationStrategy = operationStrategy;
+        this.reportGenerator = reportGenerator;
+    }
+
+    @Override
+    public void processTransactions(String inputFilePath, String outputFilePath) {
+        List<String> inputData = fileReader.read(inputFilePath);
+        List<FruitTransaction> transactions = dataConverter.convertToTransaction(inputData);
+
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.apply(transaction);
```

Comment:
The checklist recommends using the Strategy pattern without switch-cases or if-else constructions. Ensure that the `operationStrategy` uses a map of strategies to avoid such constructions and adheres to the Strategy pattern correctly.